AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/Dependency/Pinpoint/Session/ActivityTracking/StateMachine.swift

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Combine
+import Foundation
+
+class StateMachine<State, Event> {
+    typealias Reducer = (State, Event) -> State
+    private let queue = DispatchQueue(label: "com.amazonaws.Amplify.StateMachine<\(State.self), \(Event.self)>",
```

Comment:
Should we replace DispatchQueue and it usages with actor and async/await ?

Reply:
I commented about it on the PR description:

> Ended up deciding against making the client an actor as a simple DispatchQueue was sufficient for our use case: customers are not going to directly interact with this client, it will just respond to app activity.

IMO using an actor was a overkill and overcomplicated this implementation.

Reply:
My fault for missing that in the description.